### ISSUE TITLE
[openwebnet] Alarm: add a time stamp when a alarm zone event occurs 

### ISSUE THREAD

**Issue by fabgio** [openhab/openhab-addons#14535]
Channel CHANNEL_ALARM_ZONE_ALARM_STATE for an active zone should show  and persist the time of the event  

**Expected Behavior**

MM/DD/YY,HH:MM INTRUSION

**Comment by mvalla** [comment]
better define a new channel for timestamp, like is is done for Verisure binding:
https://next.openhab.org/addons/bindings/verisure/#verisure-alarm
Are you available to work on this enhancement?

**Comment by fabgio** [comment]
At the moment I am dealing with issue #14305 (debug phase) I will be testing a couple of solutions....

**Comment by mvalla** [comment]
> At the moment I am dealing with issue #14305 (debug phase) I will be debugging and testing a solution...

for that issue I just pushed PR #14566  with my proposal for a solution: feel free to test it and provide feeback commenting the PR!

**Comment by mvalla** [comment]
@fabgio are you available to work on this new channel with alarm timestamp? or should find someone else to look at this? let me know!

**Comment by fabgio** [comment]
@mvalla  Sure! I am available to work on that issue with pleasure! I will be reading some code stuff  as you suggested above!

**Comment by fabgio** [comment]
@mvalla I have pushed a commit.

**Comment by mvalla** [comment]
I gave a look [to this branch](https://github.com/openhab/openhab-addons/compare/main...fabgio:openhab-addons:14305_openwebnet_4.0.0_alarm) 
and have these comments:
- timestamp should be updated after any change to the alarm channel, independently from the new value (not just ZONE_ALARM_INTRUSION and including the alarm reset to NONE)
- similarly to Verisure Alarm binding, the channel should be of type `DateTime` (and not `String`). The new alarm value is not necessary in the channel value, because it can be read from the `alarm `channel. I suggest you look at the code in the Verisure binding to see how the channel is updated.
- `README.md` neeeds to be updated with the new channel documentation
- the new channel should be added to the .xml definitions (`channels.xml` and `BusAlarmZone.xml`). You also have to generate the default english localization, see: https://github.com/openhab/openhab-addons#translations

If you apply these changes and after you have tested successfully the new channel, I think you could submit a new PR to the main repo.

**Comment by fabgio** [comment]
ok I made all of  modifications to the code  you suggested, built, commit.  and  timestamp is correctly shown. There are only a two other problems . If the timestamp is reset as the alarm is tacitated    the user won't have enough time to read the timestamp. So  I was thinking about somekind of delayed reset of the timestamp based on the alarm event (scheduled thread?) You don't see what event (INTRUSION, TECNICAL ALARM etc..etc) the timestamp is associated.. so I am tinking  attaching a label to the ts   .....

**Comment by mvalla** [comment]
> If the timestamp is reset as the alarm is tacitated the user won't have enough time to read the notification.

I am not sure I understand: the alarm right now should be reset to NONE only after alarm is re-activated (after “delay end” frame to be more precise). So why the user should not be able to read the timestamp when, for example, alarm channel changes to INTRUSION ?

> You don't see what event (INTRUSION, TECNICAL ALARM etc..etc) the timestamp is associated so I am tinking attaching a label to the ts

I do not see any motivation to collapse the two information together in the same channel:  on the alarm channel you have the type of alarm, on the timestamp channel you have the time 

anyway, I suggest you submit a PR now so any further change suggestion can be reported in the PR review

**Comment by fabgio** [comment]
ok

**Comment by fabgio** [comment]
I am heaving difficulties in dealing with the issue. Nothing is working as expect .  I am demoralized.  Debugging is difficult because the siren starts! I am heaving regressions. I don't feel like to PR this stuff....

**Comment by mvalla** [comment]
@fabgio I looked to the branch but it seems it's not updated to latest main. Maybe try to rebase (interactive rebase) on openhab-addons/main...

**Comment by fabgio** [comment]
You are right actually  It has to be rebased on  OH 4.0! I will do it

**Comment by fabgio** [comment]
ok I updated, rebased and built  on openhab-addons/main. Here is the brench: 
https://github.com/fabgio/openhab-addons/tree/14535_openwebnet_4.0.0.

**Comment by fabgio** [comment]
I spotted an error on BusAlarmZone.xml lines 19 and 20! The channel ID alarm has got a wrong typeId!
